ocrd_models/ocrd_models/ocrd_mets.py

```diff
@@ -422,9 +614,25 @@ def set_physical_page_for_file(self, pageId, ocrd_file, order=None, orderlabel=N
         """
         #  print(pageId, ocrd_file)
         # delete any page mapping for this file.ID
-        for el_fptr in self._tree.getroot().findall(
+        
+        # NOTE: The pageId coming from 'test_merge(sbb_sample_01)' is an Element not a string
+        if not isinstance(pageId, str):
+            pageId = pageId.get('ID')
```

Comment:
```suggestion
```

This should not be the case. Are you positively certain that is the case? I cannot reproduce and if we can, that's a bug in merge we should fix.

Reply:
That was the case back then. If you cannot reproduce it anymore, then probably it's solved now. I think this error was related to the return type of `keys()` which you changed to `list(keys())` in some part of the code.

Reply:
ffcd89f37